gestor.sh

```diff
@@ -0,0 +1,337 @@
+
+	INFO "Creando $name/ (abbrev: $abbr)"
+	mkdir "$name" && cd "$name"
+
+	local extra_tex_files_input=""
+
+	# Creamos algunos directorios extra
+	for subdir in img pdf tex tikzgen; do
+		INFO "Creado directorio $name/$subdir (con fichero .keep)"
+		mkdir "$subdir"
+		touch "$subdir/.keep"
+	done
+
+	# Creamos sub-documentos por defecto
+	for suffix in Ejs; do
+		texfile="tex/${abbr}_${suffix}.tex"
+		printf -v extra_tex_files_input "$extra_tex_files_input\n\chapter{---}\n\\input{$texfile}"
+		echo "% -*- root: ../${abbr}.tex -*-" > "$texfile"
+		INFO "Creado fichero extra $texfile"
+	done
+
+	local current_year=$(date +%y)
+	local docdate
+
+	# ¿Primer o segundo cuatrimestre?
+	if [[ $(date +%m ) -ge 8 ]]; then
+		docdate="$current_year/$((current_year + 1)) C1"
+	else
+		docdate="$((current_year - 1))/$current_year C2"
+	fi
+
+	# Creamos el fichero base a partir de una plantilla
+	cat > "$abbr.tex" <<-EOF
+		\documentclass[palatino]{apuntes}
+
+		\title{$name}
+		\author{}
+		\date{$docdate}
+
+		% Paquetes adicionales
+
+		% --------------------
+
+		\begin{document}
+		\pagestyle{plain}
+		\maketitle
+
+		\tableofcontents
+		\newpage
+		% Contenido.
+
+		%% Apéndices (ejercicios, exámenes)
+		\appendix
+		$extra_tex_files_input
+		\printindex
+		\end{document}
+	EOF
+
+
+	INFO "Terminado con exito!"
+}
+
+##
+## compilar
+##
+## Compila los apuntes de la asignatura especificada.
+## Uso: compilar [asignatura]
+##
+compilar() {
+
+	# Esta separado para no ignorar el codigo de returno
+	local asignatura;
+	asignatura="$(seleccionar_asignatura "$1")"
+
+	INFO "Compilando ${F_UNDER}$asignatura${F_RESET} ..."
+
+	# Preparamos para compilar...
+	cd "$asignatura"
+	mkdir -p tikzgen
+
+	# Compilamos comprobando el codigo de returno
+	if ! latexmk -r "../latexmkrc.pl" > /dev/null; then
+
+		ERR "Ha ocurrido un error fatal al compilar"
+		ERR "Informacion de diagnostico en"
+		ERR "    ${F_UNDER}$asignatura/$(echo pdf/*.log)${F_RESET}"
+		ERR "Resuelva los errores, y vuelva a intentarlo"
+		return 1
+
+	fi
+
+	INFO "Compilado con exito!"
+	return 0
+}
+
+##
+## compilar_todo
+##
+## Compila todos los apuntes.
+##
+compilar_todo() {
+
+	local dir_num=0
+	local dir_upd=0
+	local dir_err=0
+
+	local start_time=$SECONDS
+
+	while IFS= read -r -d $'\0' asignatura; do
+
+		if ( compilar "$(dirname "$asignatura")" ); then
+			(( dir_upd += 1 ))
+		else
+			(( dir_err += 1 ))
+		fi
+
+		((dir_num += 1))
+
+		INFO "================================================================"
+
+	done < <(find . -iname '*.tex' -depth 2 -print0)
+
+	INFO "Encontrado ${F_BOLD}$dir_num${F_RESET} cursos, de los cuales"
+	INFO "    ${F_GREEN}$dir_upd${F_RESET} han sido compilados sin errores."
+	INFO "    ${F_RED}$dir_err${F_RESET} han tenido algun problema."
+	INFO ""
+	INFO "Terminado ($(date))"
+
+}
+
+##
+## limpiar
+##
+## Limpia los apuntes de la asignatura especificada.
+## Uso: limpiar [asignatura]
+##
+limpiar() {
```

Comment:
¿Has probado con asignaturas con espacios?

    $ ./gestor.sh limpiar Teoria\ de\ Galois/
    >> Limpiando Teoria de Galois ...
    Use of uninitialized value $ENV{"TEXINPUTS"} in concatenation (.) or string at (eval 8) line 15, <GEN1> chunk 1.

Con comillas tampoco funciona:

    $ ./gestor.sh limpiar "Teoria de Galois"
    >> Limpiando Teoria de Galois ...
    Use of uninitialized value $ENV{"TEXINPUTS"} in concatenation (.) or string at (eval 8) line 15, <GEN1> chunk 1.

También deberías añadir una comprobación de que hay un argumento, la he llamado sin argumentos sin saber y, obviamente, explota. Por ejemplo añadir al principio:
```bash
        if [ $# -ne 1  ]; then
                echo 'Uso: limpiar [asignatura]'
                exit -1
        fi
```
De hecho, se lo añadiría a todas las funciones que esperen argumentos para evitar problemas.